timeline-chart/src/layer/unitTest/test-data-provider.ts

```diff
@@ -0,0 +1,211 @@
+        const rows: TimelineChart.TimeGraphRowModel[] = [];
+        const rangeEvents: TimelineChart.TimeGraphAnnotation[] = [];
+
+        const chartStart = this.absoluteStart;
+
+        // Resetting
+        // this.timeGraphEntries = this.dataStub.getEntries();
+        // this.timeGraphRows = this.dataStub.getRows();
+        // this.totalLength = BigInt(0);
+
+        const rowIds = this.dataStub.getIds();
+        const orderedRows = this.timeGraphRowsOrdering(rowIds);
+
+        let totalStates = 0;
+
+        this.dataStub.getEntries().forEach((entry: any, rowIndex: number): void => {
+            let gapStyle;
+            if (!entry.style) {
+                gapStyle = this.getDefaultForGapStyle();
+            } else {
+                gapStyle = entry.style;
+            }
+
+            const states: TimelineChart.TimeGraphState[] = [];
+            const annotations: TimelineChart.TimeGraphAnnotation[] = [];
+            const row = orderedRows.find(row => row.entryId === entry.id);
+            let prevPossibleState = entry.start;
+            let nextPossibleState = entry.end;
+            if (row) {
+                row.states.forEach((state: any, stateIndex: number) => {
+                    const end = BigInt(state.end) - chartStart;
+                    states.push({
+                        id: row.entryId + '-' + stateIndex,
+                        label: state.label,
+                        range: {
+                            start: BigInt(state.start) - chartStart,
+                            end
+                        },
+                        data: {
+                            style: state.style
+                        }
+                    });
+                    //this.totalRange = this.totalRange < end ? end : this.totalRange;
+                    if (stateIndex === 0) {
+                        prevPossibleState = BigInt(state.start) - chartStart;
+                    }
+                    if (stateIndex === row.states.length - 1) {
+                        nextPossibleState = BigInt(state.end) - chartStart;
+                    }
+                });
+
+                if (states.length > 0) {
+                    totalStates += states.length;
+                    rows.push({
+                        id: entry.id,
+                        name: entry.labels[0],
+                        range: {
+                            start: BigInt(entry.start) - chartStart,
+                            end: BigInt(entry.end) - chartStart
+                        },
+                        states,
+                        annotations,
+                        data: {
+                            type: entry.type
+                        },
+                        prevPossibleState,
+                        nextPossibleState,
+                        gapStyle
+                    });
+                }
+            }
+        });
+
+        console.log("Total number of states", totalStates);

```

Comment:
space after "states"

Reply:
I removed the comment, it was a mistake.